app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.widget.TextView
+import androidx.core.graphics.toColor
+import androidx.core.widget.addTextChangedListener
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import kotlin.math.max
+import kotlin.math.min
+
+object ColorPickerUtility {
+
+    @SuppressLint("ClickableViewAccessibility")
+    fun colorPickerSheet(
+        context: Context,
+        /**
+         * Called whenever the hue is updated via text or slider.
+         * Use it to set wherever you are storing your hue
+         * and then update your color
+         */
+        onHueUpdate: (Float) -> Unit,
+        /**
+         * Called whenever the Saturation or Value is updated via the gradient square
+         * or via text. use it to set wherever you are storing your saturation AND value
+         * update your color afterward
+         */
+        onSatOrValUpdate: (Float, Float) -> Unit,
+        /**
+         * Called whenever text needs to be updated, meaning on any change, hsv, rgb, or text.
+         * Used to give the information of hsv and previous color.
+         * if there was no previous color pass it black. This is used to help
+         * populate the sheets text views and update the slider and gradient indicators as well as set the previous color box.
+         */
+        onColorTextForceUpdate: () -> DrawingViewModel.ColorPack,
+        /**
+         * Called whenever one of the RGB texts is changed by focus by a user.
+         * it passes a ColorInt as the first parameter, used to decide what value to set.
+         * the second int is the actual value you will set your R,G,or B value to.
+         * Color.RED corresponds to red,Color.GREEN to green, BLUE to blue
+         */
+        onRGBTextChange: (Int, Int) -> Unit,
+        /**
+         * called whenever the dialog is dismissed or submitted
+         * on actual submission the color will not be null so take it in and set it wherever you need the selected color and set the hsv and rgb fields to defaults.
+         * on null don't do anything with it, don't set rgb and hsv to defaults.
+         */
+        onSubmission: (Color?) -> Unit
```

Comment:
for a color gradient I wouldn't reset the hsv and rgb to defaults, a user might expect to see those remain at the values they were next time they enter the dialog, since they would be on their current color, but this is a designer question to be sure.